source/NVDAObjects/behaviors.py

```diff
@@ -366,6 +367,106 @@ def event_loseFocus(self):
 		super(Terminal, self).event_loseFocus()
 		self.stopMonitoring()
 
+
+class KeyboardHandlerBasedTypedCharSupport(Terminal):
+	"""A Terminal object that also provides typed character support for
+	console applications via keyboardHandler events.

```

Comment:
I'm still missing something fairly essential in this.

```suggestion
	console applications via keyboardHandler events.
	These events are queued from NVDA"s global keyboard hook.
	Therefore, an event is fired for every single character that is being typed,
	even when a character is not written to the console (e.g. in read only console applications).
```